src/impl_/pyclass/probes.rs

```diff
@@ -60,3 +66,19 @@ probe!(IsOption);
 impl<T> IsOption<Option<T>> {
     pub const VALUE: bool = true;
 }
+
+#[cfg(test)]
+mod value_of_impl {
+
+    #[macro_export]
+    macro_rules! value_of {
+        ($probe:ident, $ty:ty) => {{
+            #[allow(unused_imports)] // probe trait not used if VALUE is true
+            use crate::impl_::pyclass::Probe as _;
+            $probe::<$ty>::VALUE
+        }};
+    }
+}
+
+#[cfg(test)]
+pub use crate::value_of; // because `#[macro_export]` puts it in the crate root...
```

Comment:
If we only use this internally (within the `pyo3` crate), we could also write it like this and don't have to deal with `#[macro_export]` shinanigans:
```rust
#[cfg(test)]
macro_rules! value_of {
    ($probe:ident, $ty:ty) => {{
        #[allow(unused_imports)] // probe trait not used if VALUE is true
        use crate::impl_::pyclass::Probe as _;
        $probe::<$ty>::VALUE
    }};
}

#[cfg(test)]
pub(crate) use value_of;
```